src/test/java/datawave/query/util/MetadataHelperTest.java

```diff
@@ -94,7 +94,7 @@ public void testMultipleFieldFilter2() throws TableNotFoundException {
         
         Assertions.assertEquals(Collections.singleton("rowB"), helper.getAllFields(Collections.singleton("dataTypeB")));
         Assertions.assertEquals(Sets.newHashSet("rowA", "rowB", "rowC"), helper.getAllFields(null));
-        Assertions.assertEquals(Sets.newHashSet("rowA", "rowB", "rowC"), helper.getAllFields(Collections.emptySet()));
+        Assertions.assertEquals(Collections.emptySet(), helper.getAllFields(Collections.emptySet()));
```

Comment:
I thought a null datatype filter and an empty datatype filter both implied "no filter, return everything". Why then the change in behavior?

Reply:
The original intent was for a null datatype filter to allow all the things, a populated filter to allow only things contained therein, and an empty filter to allow none of the things.